src/libs/actions/Network.ts

```diff
@@ -32,4 +82,37 @@ function setShouldFailAllRequests(shouldFailAllRequests: boolean) {
     Onyx.merge(ONYXKEYS.NETWORK, {shouldFailAllRequests});
 }
 
-export {setIsOffline, setShouldForceOffline, setShouldFailAllRequests, setTimeSkew, setNetWorkStatus};
+function setPoorConnectionTimeoutID(poorConnectionTimeoutID: NodeJS.Timeout | undefined) {
+    Onyx.merge(ONYXKEYS.NETWORK, {poorConnectionTimeoutID});
+}
+
+function setRandomNetworkStatus(initialCall = false) {
+    // The check to ensure no new timeouts are scheduled after poor connection simulation is stopped
+    if (!isPoorConnectionSimulated && !initialCall) {
+        setShouldForceOffline(false);
+        return;
+    }
+
+    const statuses = [CONST.NETWORK.NETWORK_STATUS.OFFLINE, CONST.NETWORK.NETWORK_STATUS.ONLINE];
+    const randomStatus = statuses[Math.floor(Math.random() * statuses.length)];
+    const randomInterval = Math.random() * (5000 - 2000) + 2000; // random interval between 2-5 seconds
+    Log.info(`[NetworkConnection] Set connection status "${randomStatus}" for ${randomInterval} sec`);
+
+    setShouldForceOffline(randomStatus === CONST.NETWORK.NETWORK_STATUS.OFFLINE);
+
+    const timeoutID = setTimeout(setRandomNetworkStatus, randomInterval);
+
+    setPoorConnectionTimeoutID(timeoutID);
+}
+
+function simulatePoorConnection(shouldSimulatePoorConnection: boolean, poorConnectionTimeoutID: NodeJS.Timeout | undefined) {
+    if (!shouldSimulatePoorConnection) {
+        clearTimeout(poorConnectionTimeoutID);
+        setPoorConnectionTimeoutID(undefined);
+        setShouldForceOffline(false);
```

Comment:
It will be set automatically with the random value: https://github.com/Expensify/App/pull/53698/files#diff-2b29b3ab9b95c91c2073162e1518ef92c4e81388250e4a717a7ac1bd6df53760R101